### PR TITLE
Add docs for fusion strategy

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72077
* __->__ #72036
* #72038
* #72040
* #72032
* #71651
* #71650
* #71173

Differential Revision: [D33864651](https://our.internmc.facebook.com/intern/diff/D33864651)

### CODE REVIEW

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864651).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864651).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864651).

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.h [hunk 1]
```diff
@@ -7,9 +7,51 @@ namespace jit {
...
+// FusionStrategy is used to control the type and number of specializations that
+//   can occur during fusion
+//
+// Usage: provide a list of pairs (type, depth) where type is one of "STATIC" or
+//   "DYNAMIC" and depth is an integer.
+//
+// Behavior - static vs dynamic:
+// - in STATIC fusion, fused ops are compiled to have fixed input shapes. The
+//   input shapes are determined based on a number of initial profiling runs.
+//   The shape is determined based on some initial profiling runs. For example,
```
Comment (navahgar) [inline torch/csrc/jit/runtime/profiling_graph_executor_impl.h:19]: Please clarify what shape is referred to here.

**Review by navahgar — APPROVED** [review]
LGTM

File: torch/csrc/jit/runtime/profiling_graph_executor_impl.h (hunk 1 above)
Comment (eellison) [reply torch/csrc/jit/runtime/profiling_graph_executor_impl.h:19]: i think this is used throughout pytorch code base, e.g. 
image

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33864651).

**Pull request closed without merging**